cmd/natc/natc.go

```diff
@@ -201,27 +203,17 @@ func (c *connector) serveDNS(ln net.Listener) {
 //
 // This assignment later allows the connector to determine where to forward
 // traffic based on the destination IP address.
-func (c *connector) handleDNS(conn nettype.ConnPacketConn) {
-	defer conn.Close()
+func (c *connector) handleDNS(pc net.PacketConn, buf []byte, remoteAddr *net.UDPAddr) {
```

Comment:
can you make this by AddrPort instead and have caller do that?

https://pkg.go.dev/net#UDPConn.WriteToUDPAddrPort exists even if it's not in PacketConn

Reply:
I tried, but this isn't a `net.UDPConn`, this is the netstack version which is a `gonet.UDPConn` and doesn't have the `netip.AddrPort` variants.